src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -5812,7 +5812,61 @@ _pbs_python_event_set(unsigned int hook_event, char *req_user, char *req_host,
 				PY_TYPE_EVENT, PY_EVENT_PARAM_MANAGEMENT);
 			goto event_set_exit;
 		}
+	} else if (hook_event == HOOK_EVENT_MODIFYVNODE) {
+		struct rq_modifyvnode *rqmvn = req_params->rq_modifyvnode;
+		struct pbsnode *vnode_o = rqmvn->rq_vnode_o;
+		struct pbsnode *vnode = rqmvn->rq_vnode;
+
+		/* initialize event params to None */
+		(void)PyDict_SetItemString(py_event_param, PY_EVENT_PARAM_VNODE,
+			Py_None);
+		(void)PyDict_SetItemString(py_event_param, PY_EVENT_PARAM_VNODE_O,
+			Py_None);
+
+		/* Retrieve the vnode_o data */
+		py_vnode_o = _pps_helper_get_vnode(vnode_o, NULL, HOOK_PERF_POPULATE_VNODE_O);
+		if (py_vnode_o == NULL) {
+			log_err(PBSE_INTERNAL, __func__, "failed to create a python vnode_o object");
+			goto event_set_exit;
+		}
+
+		/* Set the vnode_o object to readonly to prevent hook writers from modifying values */
+		int tmpv_rc = -1;
+		tmpv_rc = pbs_python_mark_object_readonly(py_vnode_o);
+		if (tmpv_rc == -1) {
+			log_err(PBSE_INTERNAL, __func__, "Failed to mark python vnode_o object readonly");
+			goto event_set_exit;
+		}
+
+		/* Retrieve the vnode data */
+		py_vnode = _pps_helper_get_vnode(vnode, NULL, HOOK_PERF_POPULATE_VNODE);
+		if (py_vnode == NULL) {
+			log_err(PBSE_INTERNAL, __func__, "failed to create a python vnode object");
+			goto event_set_exit;
+		}
+
+		/* Set the vnode object to readonly to prevent hook writers from modifying values */
+		tmpv_rc = pbs_python_mark_object_readonly(py_vnode);
```

Comment:
As stated in the design document, this hook event is intended for notification purposes only. Making vnode read-write would require additional discussion and code modifications. Respectfully recommend such a feature change be addressed separately.

Reply:
This works for me.  It might be an interesting future enhancement when this hook event is called for more modification events than just node state changes.